### PR TITLE
Refactor error handling path in dispatching

### PR DESCRIPTION
**Pull request by chamil321**

### CODE REVIEW

File: stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/WebSubDispatcher.java [hunk 1]
```diff
@@ -36,28 +36,28 @@
...
+        HttpResource resource;
+        String protocol = (String) inboundMessage.getProperty(HttpConstants.PROTOCOL);
         if (protocol == null) {
             throw new BallerinaConnectorException("protocol not defined in the incoming request");
         }
 
         try {
             HttpService service = HttpDispatcher.findService(servicesRegistry,
-                                                             httpCarbonMessage);
+                                                             inboundMessage);
```
Comment (Bhashinee) [inline stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/WebSubDispatcher.java]: Can't we just write this in a single line? 
HttpService service = HttpDispatcher.findService(servicesRegistry, inboundMessage);

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java [hunk 2]
```diff
@@ -123,44 +123,31 @@ private static String getInterface(HTTPCarbonMessage inboundRequest) {
...
-            HttpService service = HttpDispatcher.findService(servicesRegistry, httpCarbonMessage);
+            HttpService service = HttpDispatcher.findService(servicesRegistry, inboundMessage);
             if (service == null) {
                 throw new BallerinaConnectorException("no Service found to handle the service request");
                 // Finer details of the errors are thrown from the dispatcher itself, Ideally we shouldn't get here.
             }
 
             // Find the Resource
-            resource = HttpResourceDispatcher.findResource(service, httpCarbonMessage);
+            resource = HttpResourceDispatcher.findResource(service, inboundMessage);
```
Comment (anupama-pathirage) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java]: Shall we directly return from here `return HttpResourceDispatcher.findResource(service, inboundMessage);`

File: stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/BallerinaWebSubConnectionListener.java [hunk 3]
```diff
@@ -67,35 +70,39 @@ public BallerinaWebSubConnectionListener(WebSubServicesRegistry webSubServicesRe
...
+    public void onMessage(HTTPCarbonMessage inboundMessage) {
         try {
             HttpResource httpResource;
-            if (accessed(httpCarbonMessage)) {
-                if (httpCarbonMessage.getProperty(HTTP_RESOURCE) instanceof String) {
-                    if (httpCarbonMessage.getProperty(HTTP_RESOURCE).equals(
+            if (accessed(inboundMessage)) {
+                if (inboundMessage.getProperty(HTTP_RESOURCE) instanceof String) {
+                    if (inboundMessage.getProperty(HTTP_RESOURCE).equals(
                                                                         WebSubSubscriberConstants.ANNOTATED_TOPIC)) {
```
Comment (anupama-pathirage) [inline stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/BallerinaWebSubConnectionListener.java]: Shall we combine the above two lines?

File: stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/WebSubDispatcher.java (hunk 1 above)
Comment (chamil321) [reply stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/WebSubDispatcher.java]: Ack

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java (hunk 2 above)
Comment (chamil321) [reply stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpDispatcher.java]: Ack

File: stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/BallerinaWebSubConnectionListener.java (hunk 3 above)
Comment (chamil321) [reply stdlib/ballerina-websub/src/main/java/org/ballerinalang/net/websub/BallerinaWebSubConnectionListener.java]: Ack

**Pull request merged**